### ISSUE TITLE
Soil washing into sand, gravel, and clay

### ISSUE THREAD

**Issue by Miner239** [CleverRaven/Cataclysm-DDA#75349]
### Is your feature request related to a problem? Please describe.

There are situations where you would want soil products like sand and clay, but you're unable to access pure deposits of sand and clay to extract/scavenge from. That currently leaves digging up pits (and the cheese of filling in pits and digging it back out) as the only other way to get sand and clay. This is a little ridiculous as soil itself is partly made out of sand. Washing soil should be a possible way to get sand.

Taking reference from [Profile Characteristics of New England Forest Soils](https://portal.ct.gov/-/media/caes/documents/publications/bulletins/b342pdf.pdf), New England soil is mostly sand, with silt and clay content ranging from 5-30%. I picked sample 37 as representative of in-game soil for its proximity to Massachusetts and the river. 

The sample, after removing pebbles and rocks larger than a quarter inch and dried, is composed of 5.1% volatiles, 67.3% coarse sand, 3.5% silt, and 7.7% clay, adding up to 83.6%. Adding back in 5.18% moisture content gives us a total of 88.78%. 

The unaccounted for 11.22% fraction is assumed to be, and equally divided into 2mm-6mm gravel and fine sand as the categories missing in the table. I also assume that the gravel and small pebbles the source removed for testing purposes are present in our soil and count as in-game gravel. Since silt is not implemented in game and does not have any meaningfully different use worthy of adding it right now, I also divide it to add to sand and clay. 

This gives us a mass yield percentage of 5.61% gravel, 74.66% sand, and 9.45% clay. Translating it to C:DDA units, with a slight nudge of numbers, gives us 46 gravel, 607 sand, and 1 clay for every 1 soil.
![image](https://github.com/user-attachments/assets/9501797f-eca0-41d5-9c13-cae9b46a10e0)

### Solution you would like.

Implement a set of recipe that puts in soil and water, and gives out sand, with various byproducts. Craft times and water amounts are tentative.

Base:
- 1 (6.5kg) soil + 40 (10L) water + containing 1 tool + 30 minutes of vigorous manual washing/panning/pouring -> 46 (368g) gravel + 607 (4856g) sand + 1 (10L) clay water
- 1 clay water transforms into 1 wet clay and 39 murky water in 2 hours
- 1 wet clay transforms into 1 clay in 1 day

Variations:
- base recipe + sieving 1 tool takes 3 minutes of washing on a sieve instead of 30 minutes
- Clay water + straining 2 tool crafts wet clay faster than waiting for it to settle/transform
- Wet clay + surface heat dries it faster?

### Describe alternatives you have considered.

- Shrink clay/add smaller clay pieces to allow clay water to be smaller than 10 liter and fit in more containers. Then clay crafting recipes and clay producers also needs editing to match new stats
- Shrink soil. idem.
- Implement silt and find where it can be used/substituted into instead of mixing it in with sand and clay
- Not assume there's gravel mixed in with the soil
- Pick the average of the data for the representative of New England soil composition

**Comment by kevingranade** [comment]
I want to dig into `There are situations where you would want soil products like sand and clay, but you're unable to access pure deposits of sand and clay to extract/scavenge from.` a bit.
What situations? what do you need the soil products for? Is this because of incomplete mapgen or what?

Basically, the preferred solution for this should be "go find sand/gravel/clay", that is what people actually do in this situation, washing soil to separate it is theoretically possible but incredibly time intensive.

**Comment by Miner239** [comment]
The quickest example I can think of where you might not have enough safe land area to scavenge from is island starts, but scenarios like prison or middle-of-city starts also applies. Or the player might just not have the logistics to find or move them out of the location once extracted. Soil products are heavy, so quarrying the local terrain for products should be a viable option.

The purpose... well, I've been reading up on terrain construction numbers and found a lot of them require way less than they should be, so I also intend to go through all of them and set them to proper numbers. 7 liters of rock to make a complete stone wall? FOUR liters of dry concrete mix to make a reinforced concrete wall?? If the wall is a square meter, it would be 4 mm thick lol. And I heard that roof-holding walls in CDDA are assumed to be 4 m high, so that only makes the age-old costs even more ridiculous. After that issue is fixed, construction work would need a lot more than it costs now, some a hundred times more than it currently is. That's the purpose I want to give local quarrying a chance for.

I'll collect the terrain construction costs and post it as a separate issue within the week.

**Comment by kevingranade** [comment]
> The quickest example I can think of where you might not have enough safe land area to scavenge from is island starts

Ok and then you have tools to do this with because??? And what use cases?  Just because something's an island doesn't mean it's just uniform soil, on the contrary they're even more likely to have sand deposits, or worse just sand deposits.  Basically you're saying an island large enough to maintain a normal loam mix but too small to have clay or sand deposits?

> scenarios like prison or middle-of-city starts also applies

In which case you're getting the massive amounts of water needed by this process from... where?

> Or the player might just not have the logistics to find or move them out of the location once extracted

Like what, a wheelbarrow?

> Soil products are heavy, so quarrying the local terrain for products should be a viable option.

That's backwards, soil products are heavy and incredibly difficult to process, so it *isn't* a viable option.  Yes they take a lot of work to transport, but they take *way more* to process.  The only time people do stuff like this is either massive industrial scale, quality control or chasing after resources that are incredibly hard to come by like precious metals.

> terrain construction numbers

You want this for bulk terrain construction??? That's so much worse!  If you wanted just a little bit it'd be one thing, but massive amounts of raw material for building construction really, really indicates you need to go harvest it somewhere instead of separating.

Do the math, I think your estimates are low for washing time, but even then, the production rate is not going to be able to keep up with what you want for construction.  You want hundreds and hundreds of pounds of material, and grinding that out a few kg at a time is not going to be great.  If the result is "but that means I can't do my construction project in this weird location with no access to sand", that's totally fine.

Also silt+clay isn't clay? Your numbers end up at about 33% silt 66% clay, which is considered clay from a soil composition point of view, but I suspect it's a very poor representation of what we want clay for, which is dominated by pottery.

**Comment by Miner239** [comment]
> Do the math, I think your estimates are low for washing time, but even then, the production rate is not going to be able to keep up with what you want for construction. You want hundreds and hundreds of pounds of material, and grinding that out a few kg at a time is not going to be great. If the result is "but that means I can't do my construction project in this weird location with no access to sand", that's totally fine.

My estimates are actually very pessimistic, and even my guess was too much.

I went ahead and dug up around 6L of my local soil, which is almost pure clay (I couldn't find any research on the topic, but the result shows itself), and used 10L of water to wash and dissolve the clay from the tiny amount of gravel and sand left behind. After pouring in 5L of water, I cleaned out the large debris and kneaded the solids underwater until I can feel a clay layer settling and making that water layer heavier than the water above it when I pause for a few seconds. This took around 5 minutes. Drain the clay water while sieving the water with my hand to catch the smaller debris, pour in 1L, repeat kneading. After I used up 8L, it has mostly turned into finding the stubborn pieces of clay and breaking them up to dissolve, the gravel and sand have been mostly cleaned. 

![image](https://github.com/user-attachments/assets/5219221d-3c69-456d-bb38-d71bf45eb757)
(6L of soil was around half of the black bucket)

The washing took around 30 minutes total. With a sand-majority soil, this would have been much faster since there's less clay to dissolve. The clay water has already settled a little in that 30 minutes, as you can see in the picture, and you can take that murky water and reuse it to wash more soil.

I haven't tested using a sieve, though. 

Someone in NE should test this and see how fast they can clean that much soil, it only takes an hour to dig and wash. With much less clay content to wash away, you should be able to use less water and time to do it.

> Also silt+clay isn't clay? Your numbers end up at about 33% silt 66% clay, which is considered clay from a soil composition point of view, but I suspect it's a very poor representation of what we want clay for, which is dominated by pottery.

It ends up at (.035/2)/((.035/2)+.077)=18.52%. Or we can ignore silt completely and make it disappear, the recipe will still yield 1 clay all the same, but with a bit less sand. Which solution do you prefer?

> Ok and then you have tools to do this with because??? 

Wooden utensils should be enough. Wooden bowls/pots. Or pottery.

> And what use cases? 

In small amounts, clay for pottery, sand for molds and mortar.

> Just because something's an island doesn't mean it's just uniform soil, on the contrary they're even more likely to have sand deposits, or worse just sand deposits. Basically you're saying an island large enough to maintain a normal loam mix but too small to have clay or sand deposits?

That's one case, yes. I also just remembered that it'll be quite useful for Innawoods players.

> In which case you're getting the massive amounts of water needed by this process from... where?

Pools, Rainwater. Toilets. The amount of water you have limits how much clay is settling at the same time, you can reuse the water after the clay settles.

> Like what, a wheelbarrow?

> That's backwards, soil products are heavy and incredibly difficult to process, so it _isn't_ a viable option. Yes they take a lot of work to transport, but they take _way more_ to process. The only time people do stuff like this is either massive industrial scale, quality control or chasing after resources that are incredibly hard to come by like precious metals.

Or a cart and some bags, or a truck. But that requires the player to obtain all that first, Yes, even relatively modest logistic infrastructure does make the transport labor less than refining local soil. But then again, 

> You want this for bulk terrain construction??? That's so much worse! If you wanted just a little bit it'd be one thing, but massive amounts of raw material for building construction really, really indicates you need to go harvest it somewhere instead of separating.

**Comment by kevingranade** [comment]
I appreciate your initiative, but primary research isn't a thing for this.  What you have here is a picture of gravel in a bucket and some muddy water, there are too many questions left to count and I'm not getting into it.  Just to mention the top issue, do you really live in an area with like 0.001% sand/gravel content, or is the washing just incomplete and your "clay bucket" is actually just "everything but the gravel"?

This relates to you dismissing the clay/silt issue. *The reason workable clay is a pain to find* is that amending it to be pure clay is difficult and not worth doing, so you just keep looking until you find a good enough clay source. That's what you do because amending soil is HARD.

> Wooden utensils should be enough. Wooden bowls/pots. Or pottery.

So now you're one, locking pottery behind wood carving *or pottery*, and two dropping the batch size to well under a liter.  Do you not see how this is not working?

Bottom line, you're not presenting anything convincing me this is worth adding.

**Comment by Miner239** [comment]
You're right, it's actually half silt. Using cloth does make it trivial to separate them. The clay I did manage to separate still hasn't dried in the week since since it's been raining all the time here.

Pottery is already locked behind wood carving, requiring digging 1 to dig clay. I don't see what's wrong with adding a more passive path to get more clay once you've got some bootstrap pottery going.

The bowls/pots are just the tool to wash the soil on, the batch size is still one soil per craft. Unless you want me to also reduce soil and clay unit size to match the rest? That's a good alternative, sure.

What will convince you to let me add it?

**Comment by kevingranade** [comment]
Nothing, because it's a nonsensical thing to do.  If you need clay/sand/gravel you go find it, you don't process raw dirt into usable quantities of any of the above.
